src/base/datetime/calendar/header/dropdown/year/index.ts

```diff
@@ -0,0 +1,227 @@
+import { BaseDateTimeListbox, Item } from "../../../../listbox";
+import { getToggleIconSvgTemplate } from "../../ultils";
+
+export class BaseDateTimeYearDropdown extends KucBase {
+  @property({ type: Number }) year = 2021;
+  @property({ type: String }) postfix = "";
+
+  @state()
+  private _menuVisible = false;
+
+  @state()
+  private _currentYear = new Date().getFullYear();
+
+  private _GUID = generateGUID();
+  private _menuItems: Item[] | undefined;
+
+  @query(".kuc-base-datetime-year-dropdown__toggle")
+  private _toggleEl!: HTMLButtonElement;
+
+  @query(".kuc-base-datetime-year-dropdown__listbox")
+  private _menuEl!: BaseDateTimeListbox;
+
+  update(changedProperties: PropertyValues) {
+    this._menuItems = this._getYearOptions().map((year: number) => {
+      const item: Item = {
+        value: `${year}`,
+        label: `${year}${this.postfix}`
+      };
+      return item;
+    });
+    super.update(changedProperties);
+  }
+
+  render() {
+    return html`
+      ${this._getStyleTagTemplate()}
+      <button
+        class="kuc-base-datetime-year-dropdown__toggle"
+        aria-haspopup="true"
+        aria-labelledby="${this._GUID}-label ${this._GUID}-toggle"
+        @mouseup="${this._handleMouseUpDropdownToggle}"
+        @mousedown="${this._handleMouseDownDropdownToggle}"
+        @click="${this._handleClickDropdownYearToggle}"
+        @blur="${this._handleBlurDropdownYearToggle}"
+        @keydown="${this._handleKeyDownYearToggle}"
+      >
+        <span class="kuc-base-datetime-year-dropdown__toggle__label"
+          >${this.year}</span
+        >
+        <span class="kuc-base-datetime-year-dropdown__toggle__icon"
+          >${getToggleIconSvgTemplate()}
+        </span>
+      </button>
+      <kuc-base-datetime-listbox
+        .items="${this._menuItems || []}"
+        .value="${this.year.toString()}"
+        class="kuc-base-datetime-year-dropdown__listbox"
+        @kuc:calendar-listbox-click="${this._handleChangeMenu}"
+        aria-hidden="${!this._menuVisible}"
+        ?hidden="${!this._menuVisible}"
+      >
+      </kuc-base-datetime-listbox>
+    `;
+  }
+
+  private _getStyleTagTemplate() {
+    return html`
+      <style>
+        .kuc-base-datetime-year-dropdown__toggle {
+          position: relative;
+          box-sizing: border-box;
+          height: 32px;
+          padding: 0 14px 0 8px;
+          line-height: 30px;
+          overflow: hidden;
+          background-color: white;
+          border: 1px solid transparent;
+          cursor: pointer;
+        }
+        .kuc-base-datetime-year-dropdown__toggle__icon {
+          flex: none;
+          width: 38px;
+          height: 38px;
+        }
+      </style>
+    `;
+  }
+
+  private _handleMouseUpDropdownToggle(event: MouseEvent) {
+    event.preventDefault();
+  }
+
+  private _handleMouseDownDropdownToggle(event: MouseEvent) {
+    event.preventDefault();
+  }
+
+  private _handleClickDropdownYearToggle(event: MouseEvent) {
```

Comment:
Delete the unnecessary MouseEvent.
```suggestion
  private _handleClickDropdownYearToggle() {
```